display/ssd1306.h

```diff
@@ -468,26 +486,63 @@ class StandardDisplayController : public DisplayControllerBase<Width, col_t>, Sa
 	  ShowFile(&IMG_idle, 3600000.0);
 	}
 	return;
-      case EFFECT_BLAST:
-	ShowFile(&IMG_blst, font_config.ProffieOSBlastImageDuration);
-	return;
-      case EFFECT_CLASH:
-	ShowFile(&IMG_clsh, font_config.ProffieOSClashImageDuration);
-	return;
-      case EFFECT_FORCE:
-	ShowFile(&IMG_force, font_config.ProffieOSForceImageDuration);
-	return;
       case EFFECT_LOCKUP_BEGIN:
-	ShowDefault();
-	return;
+      	ShowDefault();
+      	return;
       case EFFECT_LOCKUP_END:
-	ShowDefault(true);
-	break;
+      	ShowDefault(true);
+      	break;
+
+      default: break;
+    }
+  }
+
 
+ void SB_Effect2(EffectType effect, float location) override {
+    switch (effect) {
+      case EFFECT_BLAST:
+        ShowFile(&IMG_blst, font_config.ProffieOSBlastImageDuration);
+        return;
+      case EFFECT_CLASH:
+        ShowFile(&IMG_clsh, font_config.ProffieOSClashImageDuration);
+        return;
+      case EFFECT_FORCE:
+        ShowFile(&IMG_force, font_config.ProffieOSForceImageDuration);
+        return;
+      case EFFECT_PREON:
+        ShowFile(&IMG_preon, ProffieOSPreonImageDuration;);                                                                                                // BC another one that doesn't like float....?
```

Comment:
This seems untested, because this won't compile.
(You have a stray semicolon in there..)